pep-0572.rst

```diff
@@ -615,12 +615,36 @@ Broadly the same semantics as the current proposal, but spelled differently.
 
        stuff = [[f(x) as y, x/y] for x in range(5)]
 
-   Since ``EXPR as NAME`` already has meaning in ``except`` and ``with``
-   statements (with different semantics), this would create unnecessary
-   confusion or require special-casing (eg to forbid assignment within the
-   headers of these statements).
+   Since ``EXPR as NAME`` already has meaning in ``import``,
+   ``except`` and ``with`` statements (with different semantics), this
+   would create unnecessary confusion or require special-casing
+   (e.g. to forbid assignment within the headers of these statements).
 
-   TODO: Explain more why this is bad, since this keeps coming up (it's also readability)
+   Additional reasons to prefer ``:=`` over this spelling include:
+
+   - In ``if f(x) as y`` the assignment target doesn't jump out at you
+     -- it just reads like ``if f x blah blah`` and it is too similar
+     visually to ``if f(x) and y``.
+
+   - In all other situations where an ``as`` clause is allowed, even
+     readers with intermediary skills are led to anticipate that
+     clause (however optional) by the keyword that starts the line,
+     and the grammar ties that keyword closely to the as clause:
+
+     - ``import foo as bar``
+     - ``except Exc as var``
+     - ``with ctxmgr() as var``
+
+     To the contrary, the assignment expression does not belong to the
+     ``if`` or ``while`` that starts the line, and we intentionally
+     allow assignment expressions in other contexts as well.
+
+   - The parallel cadence between
+
+     - ``NAME = EXPR``
+     - ``if NAME := EXPR``
+
+     reinforces the visual recognition of assignment expressions.
```

Comment:
And there's a rhyming cadence in that `for` loop headers also put the targets to the left and the values to the right.